### PR TITLE
PR - User Edit Form (Incomplete but working)

### PR DESCRIPTION
**Pull request by rfontanarosa**
part of #2024

What have been done:

- Changed User type
- User migration
- Moved name, lang, institution and position under user.props
- used user.props.name and user.props.lang where necessary
- Full working UserList
- Full working UserForm (now supports user property edit)

### CODE REVIEW

File: src/meta/user/user.ts [hunk 1]
```diff
@@ -6,15 +6,23 @@ export enum UserStatus {
   inactive = 'inactive',
 }
 
+export type UserProps = {
+  // old fields
+  name: string
+  lang: string
+  institution?: string
+  position?: string
```
Comment (minotogna) [inline src/meta/user/user.ts:14]: to remove. in migration they will become role props (organization, position)

File: src/meta/user/userRole.ts [hunk 2]
```diff
@@ -9,7 +9,30 @@ export enum RoleName {
   VIEWER = 'VIEWER',
 }
 
-export interface UserRole<N extends RoleName, P = undefined> {
+export type UserRoleProps = {
+  userInfo: {
```
Comment (minotogna) [inline src/meta/user/userRole.ts:13]: remove userInfo object. create hierarchy

File: src/meta/user/user.ts (hunk 1 above)
Comment (rfontanarosa) [reply src/meta/user/user.ts:14]: ok, so they will be replicated for every role of the user. but position will be Professional title / Titre professionnelle / Título professional

Comment (rfontanarosa) [reply src/meta/user/user.ts:14]: next pr

**Review by minotogna — COMMENTED** [review]
Pr too big. Can we split first BE changes and then FE ?

**Pull request closed without merging**